Exception during error handling

We are running ansible-cvp v1.1.2, with CVP 2019.1.4.

We have an exception that is raised when a CvpApiError is handled during cv_configlet invocation :
```
/var/lib/awx/venv/cheops-py3/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host 'xxx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/var/lib/awx/venv/cheops-py3/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host 'xxx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/var/lib/awx/venv/cheops-py3/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host 'xxx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/var/lib/awx/venv/cheops-py3/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host 'xxx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
/var/lib/awx/venv/cheops-py3/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host 'xxx'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py\", line 344, in action_update
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_api2019.py\", line 772, in update_configlet
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 759, in post
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 537, in _make_request
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 533, in _make_request
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 684, in _send_request
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 652, in _send_request
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/cv_client.py\", line 423, in _is_good_response
ansible_collections.arista.cvp.plugins.module_utils.cv_client_errors.CvpApiError: POST: https://xxx:443/web/configlet/updateConfiglet.do : Request Error: Access Denied

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"/var/lib/awx/.ansible/tmp/ansible-local-1309927efk825k/ansible-tmp-1599055547.4221933-665-26587606830784/AnsiballZ_cv_configlet.py\", line 102, in <module>
    _ansiballz_main()
  File \"/var/lib/awx/.ansible/tmp/ansible-local-1309927efk825k/ansible-tmp-1599055547.4221933-665-26587606830784/AnsiballZ_cv_configlet.py\", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/var/lib/awx/.ansible/tmp/ansible-local-1309927efk825k/ansible-tmp-1599055547.4221933-665-26587606830784/AnsiballZ_cv_configlet.py\", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.arista.cvp.plugins.modules.cv_configlet', init_globals=None, run_name='__main__', alter_sys=True)
  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code
    exec(code, run_globals)
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py\", line 714, in <module>
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py\", line 706, in main
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py\", line 658, in action_manager
  File \"/tmp/ansible_arista.cvp.cv_configlet_payload__375u7cg/ansible_arista.cvp.cv_configlet_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py\", line 350, in action_update
KeyError: 'name'
```
This offuscates the real error.

Dear @netixx 

I am truying to replicate your issue but I am not able to get exception you have in this output.
Could you elaborate more about Radius state to get such message ?

For information, here is what I've got from ansible using `v1.1.2`

- Invalid user:

```shell
$ ansible-playbook playbooks/extract-facts.yml --extra-vars "output_file=../cvp-debug-logs/arista.cvp.facts.json" -i inventories/emea-cvp-dc1/inventory.yml

PLAY [CV Facts] ******

TASK [Gather CVP facts from cv_server] ******
Wednesday 09 September 2020  10:31:30 +0200 (0:00:00.062)       0:00:00.062 ***
Wednesday 09 September 2020  10:31:30 +0200 (0:00:00.061)       0:00:00.061 ***
fatal: [cv_server]: FAILED! => changed=false
  msg: |2-

    x.x.x.x: Authenticate: https://x.x.x.x:443/web/login/authenticate.do: Request Error: Invalid credentials
```

- Server unreachable

```
ansible-playbook playbooks/extract-facts.yml --extra-vars "output_file=../cvp-debug-logs/arista.cvp.facts.json" -i inventories/emea-cvp-dc1/inventory.yml

PLAY [CV Facts] ****************************************************************************************************************************************************************************************************************************************************************

TASK [Gather CVP facts from cv_server] *****************************************************************************************************************************************************************************************************************************************
Wednesday 09 September 2020  10:21:17 +0200 (0:00:00.086)       0:00:00.086 ***
Wednesday 09 September 2020  10:21:17 +0200 (0:00:00.085)       0:00:00.085 ***
fatal: [cv_server]: FAILED! => changed=false
  msg: |2-

    x.x.x.x: Authenticate: https://x.x.x.x:443/web/login/authenticate.do: Request Error: Server unreachable
[cv_server] TASK: Gather CVP facts from cv_server (debug)> make: *** [facts] Error 99
```

--- a/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py
+++ b/ansible_collections/arista/cvp/plugins/modules/cv_configlet.py
@@ -321,23 +321,26 @@ def action_update(update_configlets, module):
             flag_failed = True
             # Build error message to report in ansible output
             errorMessage = re.split(':', str(error))[-1]
-            message = "Configlet %s cannot be updated - %s" % (configlet['name'], errorMessage)
+            errorKey = 'Undefined Configlet'
+            if 'name' in configlet['data'].keys():
+                errorKey = configlet['data']['name']
+            message = "Configlet %s cannot be updated due to cv_client exception - %s" % (errorKey, errorMessage)
             # Add logging to ansible response.
-            response_data.append({configlet['name']: message})
+            response_data.append({errorKey:errorMessage})
             # Generate logging error message
-            MODULE_LOGGER.error('Error updating configlet %s: %s', str(
-                configlet['data']['name']), str(error))
+            MODULE_LOGGER.error('Exception raised when updating configlet %s: %s', str(
+                errorKey), str(errorMessage))
         else:
             MODULE_LOGGER.debug('CV response is %s', str(update_resp))
             if "errorMessage" in str(update_resp):
                 # Mark module execution with error
                 flag_failed = True
                 # Build error message to report in ansible output
-                message = "Configlet %s cannot be updated - %s" % (configlet['name'], update_resp['errorMessage'])
+                message = "Configlet %s cannot be updated due to a CV error- %s" % (configlet['data']['name'], update_resp['errorMessage'])
                 # Add logging to ansible response.
                 response_data.append({configlet['data']['name']: message})
                 # Generate logging error message
-                MODULE_LOGGER.error('Error updating configlet %s: %s', str(
+                MODULE_LOGGER.error('Error on CV updating configlet %s: %s', str(
                     configlet['data']['name']), str(update_resp['errorMessage']))
             else:
                 # Inform module a changed has been done

